app/controllers/api/users_controller.rb

```diff
@@ -2,98 +2,74 @@ class Api::UsersController < ApplicationController
   skip_before_action :authenticate_request, only: %i[login register]
 
   def login
-    # TODO: Don't use user params directly, always sanitize user input!
-    # Here: use user_params instead of params
-    authenticate params[:name], params[:password]
+    if authenticator.invalid?
+      render json: {errors: authenticator.errors}, status: :unprocessable_entity 
+    else
+      if authenticator.authenticated
+        user = authenticator.user
+        render json: {
+          user: {
+            id: user.id,
+            name: user.name,
+            access_token: authenticator.token
+          }
+        }, status: :ok
+      else
+        head :forbidden
+      end
+    end
   end
 
-	def index
-    # TODO: I guess this is a placeholder for now, right?
-		i = {"user" => "Hello world!"}
-		render json: i
-	end
-
   def register
-    @user = User.create(user_params)
+    @user = User.new(user_params)
     if @user.save
-      response = { message: 'User created successfully' }
+      authenticator.authenticated
```

Comment:
This call should be moved to the if condition.
It's very unprobable, but it somehow might happen that this function returns false, in which case you want to render an error status

Reply:
Please update this